### DISCUSSION TITLE
v1.9.0

### DISCUSSION THREAD

**Discussion by marcus-crane** [marcus-crane/october#128] [Announcements]
This release introduces a command line interface (CLI) for power users of October, who prefer to sync highlights without opening the desktop UI.

Users of automation software may also find this handy for automatically triggering a sync when their automation software detects that a Kobo is connected.

At the moment, it just has basic sync functionality with settings still powered by the UI but eventually it'll be brought up to feature parity with the rest of the desktop UI.

![cli-macos](https://github.com/marcus-crane/october/assets/14816406/d5750677-41ad-43b9-972f-9a0a617dedc3)

This release also brings official support for the following newer devices:

- Clara 2E
- Elipsa 2E
- Libra Colour
- Clara BW
- Clara Colour

Functionally, they should have always worked but the UI will now recognise their model names.

## What's Changed
* Pin wails and upgrade Go to 1.21 by @marcus-crane in https://github.com/marcus-crane/october/pull/112
* Only fetch user library when upload cover setting is enabled by @marcus-crane in https://github.com/marcus-crane/october/pull/115
* Introduce an initial attempt at a CLI by @marcus-crane in https://github.com/marcus-crane/october/pull/113
* Sign binary before adding to macOS app by @marcus-crane in https://github.com/marcus-crane/october/pull/116
* [SIGN] Test codesigning during pull requests by @marcus-crane in https://github.com/marcus-crane/october/pull/117
* [SIGN] Attempt to package up Windows CLI by @marcus-crane in https://github.com/marcus-crane/october/pull/119
* v1.9.0: Write documentation for CLI tool by @marcus-crane in https://github.com/marcus-crane/october/pull/127

**Full Changelog**: https://github.com/marcus-crane/october/compare/v1.8.0...v1.9.0

<hr /><em>This discussion was created from the release <a href='https://github.com/marcus-crane/october/releases/tag/v1.9.0'>v1.9.0</a>.</em>

**Comment by GeorgeHahn** [comment]
Exciting! This is getting very close to something that could run on-device. I've mostly switched over to a Supernote but I'd still like to try building this for Kobo at some point.

**Reply by marcus-crane** [reply]
Oh, I suppose it could run as some sort of systemd/init.d service as far as just the CLI goes, that's true! Personally I use a Boox Tab Mini C with Readwise Reader on it myself haha.

Perhaps at that point, something a little more lightweight would make more sense though 🤔

**Comment by Dulc1gno** [comment]
Hello,

trying to run the latest version on my m1 Mac with a new Kobo Clara BW. "No devices found"

**Reply by marcus-crane** [reply]
Hi @Dulc1gno,

I would recommend filing an issue to track issues in future (it just makes them easier for me to follow and update) but here is fine.

Hmm, there's always the chance that new Kobo devices have some slight changes given that I don't have any special access to Kobo devices and it's not really reasonable for me to just buy every device that comes out so I'll need a bit more information.

When your Kobo is connected, where does it get mounted at? For macOS, I would expect it to appear at `/Volumes/KOBOeReader` when selecting the device in the Finder sidebar? That's the first thing to cross off the list and then we can look at next steps

**Reply by Dulc1gno** [reply]
Thanks for your response. Will do. 

I have seemed to figure out that it would not recognize my renamed device so I just used the original device name and it was found. 

However, I am getting this error message when trying to sync highlights after verifying my API key.

"There was a problem sending your highlights: received a non-200 status code from Readwise: code 400"

**Reply by marcus-crane** [reply]
@Dulc1gno When you say "renamed device", do you mean that you changed `/Volumes/KOBOeReader` to something else like `/Volumes/Blah`?

Ah, as for that 400 issue, I still need to fix that bug but for now, you should be unblocked if you enable "Sync store-purchased books" in Settings after selecting a device (even though you probably don't have any)

**Reply by Dulc1gno** [reply]
Yes, exactly. 

Enabled. Same error.
